server.js

```diff
@@ -27,9 +19,84 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// set up models (Schema) (define what type of data)
+// you could put all models in a models-folder also.
+const NetflixTitle = new mongoose.model('NetflixTitle', {
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: Number,
+  rating: String,
+  duration: String,
+  listed_in: String,
+  description: String,
+  type: String
+})
+
+//Populate the database
+if (process.env.RESET_DATABASE) { // environment variable used to reset the database
+  const seedDatabase = async () => {
+    await NetflixTitle.deleteMany(); //main rule: clean database before populating database.
+
+    netflixData.forEach(item => {
+      const newNetflixTitle = new NetflixTitle(item); // item is a single object from the json array.
+      newNetflixTitle.save();
+    })
+  }
+  seedDatabase();
+}
+
+//ENDPOINT TO GET ALL TITLES: '/netflix-titles'
+//QUERY-PARAM TO GET ONE TITLE BY TITLE-NAME: /netflix-titles?title=American 
+//QUERY-PARAM TO GET DIRECTOR: /netflix-titles?director=nate
+//Query-PARAM TO GET ACTOR: /netflix-titles?actor=niro
+//Query-PARAM TO GET TITLES BY TYPE: /netflix-titles?type=tv
+//Query-PARAM TO GET TITLES FROM COUNTRY: /netflix-titles?country=germany
+//EXAMPLE TO FIND TITLE(S) BY DIRECTOR & ACTOR /netflix-titles?actor=niro&director=scorsese
+app.get('/netflix-titles', async (req, res) => {
+  const { title, director, actor, type, country } = req.query;
+  const allTitles = await NetflixTitle.find({
+    title: new RegExp(title, 'i'), //regexp: 'i' is 'ignore case'
+    director: new RegExp(director, 'i'),
+    cast: new RegExp(actor, 'i'),
+    type: new RegExp(type, 'i'),
+    country: new RegExp(country, 'i')
+  })
+    .sort({ release_year: -1 }); //sort desc.
+  if (allTitles.length === 0) {
+    res.status(404).json({ error: 'Could not find any Netflix-titles to show' })
+  } else {
+    res.json(allTitles);
+  }
+})
+
+//YEAR-ENDPOINT: get all titles from a specific year.
+app.get('/netflix-titles/year/:year', async (req, res) => {
```

Comment:
I think this could have been a query too for consistency maybe? 
And if you really wanted this to be an endpoint consider making it into 
`app.get('/netflix-titles/:year', async (req, res)`
To make it a bit more easy for the user

Reply:
when you are filtering anything else then `id` you need to add the name of a filtered field to the path so `/netflix-titles/year/:year` is the good endpoint.